### PR TITLE
loop and transform_loop unseq adaptation

### PR DESCRIPTION
**Pull request by akcube**
## Proposed Changes
- Implements the unseq adaptation of `util::loop` and `transform_loop`. Will affect `transform`, `for_each` and `for_loop`.
- Added tests for both.

## Any background context you want to provide?
This is based on #6016

### CODE REVIEW

File: libs/core/algorithms/include/hpx/parallel/unseq/loop.hpp [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+            template <typename Begin, typename End, typename F>
+            HPX_HOST_DEVICE HPX_FORCEINLINE static typename std::enable_if<
+                !hpx::traits::is_random_access_iterator_v<Begin>, Begin>::type
+            call(Begin it, End end, F&& f)
+            {
+                for (/* */; it != end; it++)
+                {
+                    HPX_INVOKE(f, it);
+                }
+            }
```
Comment (hkaiser) [inline libs/core/algorithms/include/hpx/parallel/unseq/loop.hpp:102]: Could we make this fall back to the existing non-vectorized implementation?

File: libs/core/algorithms/include/hpx/parallel/unseq/loop.hpp [hunk 2]
```diff
@@ -0,0 +1,356 @@
...
+                F&& f)
+            {
+                for (/* */; it != end; ++it)
+                {
+                    HPX_INVOKE(f, it);
+                    if (tok.was_cancelled())
+                        return it;
+                }
+                return it;
+            }
```
Comment (hkaiser) [inline libs/core/algorithms/include/hpx/parallel/unseq/loop.hpp:129]: Same here: Could we make this fall back to the existing non-vectorized implementation?

File: libs/core/algorithms/include/hpx/parallel/unseq/loop.hpp [hunk 3]
```diff
@@ -0,0 +1,356 @@
...
+            HPX_HOST_DEVICE HPX_FORCEINLINE static typename std::enable_if<
+                !hpx::traits::is_random_access_iterator_v<Begin>, Begin>::type
+            call(Begin it, End end, F&& f)
+            {
+                for (/* */; it != end; ++it)
+                {
+                    HPX_INVOKE(f, *it);
+                }
+                return it;
+            }
```
Comment (hkaiser) [inline libs/core/algorithms/include/hpx/parallel/unseq/loop.hpp:153]: Could we make this fall back to the existing non-vectorized implementation? This comment applies to many more places in this file below.

**Review by hkaiser — COMMENTED** [review]
I think we could reduce the amount of code duplication in loop.hpp et.al. by enabling the tag_invoke overloads for unseq  for random_access iterators only.

**Comment by Johan511** [comment]
@akcube  can you please help me out with how I can run the algorithms using the par_unseq execution policy?  I am not able to be unseq/loop.hpp in my hpx installation

`sudo find . -name "*loop.hpp*" -print`

`./include/parallel_for_loop.hpp`
`./thread_pools/scheduling_loop.hpp`
`./execution/algorithms/run_loop.hpp`
`./parallel/container_algorithms/for_loop.hpp`
`./parallel/util/transform_loop.hpp`
`./parallel/util/loop.hpp`
`./parallel/algorithms/for_loop.hpp`
`./parallel/datapar/transform_loop.hpp`
`./parallel/datapar/loop.hpp`

also I did place a breakpoint at loop.hpp and did find the execution breaking over [here](https://github.com/STEllAR-GROUP/hpx/blob/master/libs/core/algorithms/include/hpx/parallel/util/loop.hpp#L302). Was this the expected behaviour?

**Pull request closed without merging**